resources/web/wwi/RobotWindow.js

```diff
@@ -3,8 +3,7 @@ import {getGETQueryValue} from './request_methods.js';
 export default class RobotWindow {
   constructor(onready) {
     this.name = decodeURI(getGETQueryValue('name', 'undefined'));
-    this.wsServer = decodeURI(getGETQueryValue('url', 'ws://localhost:1234/'));
-
+    this.wsServer = window.location.href.substring(0, window.location.href.indexOf('robot_windows')).replace('https', 'wss').replace('http', 'ws');
```

Comment:
We should be super careful with this string replacements as it introduces weaknesses in the code. For example a stronger version would be:
```js
this.wsServer = window.location.href.substring(0, window.location.href.indexOf('robot_windows')).replace('https://', 'wss://').replace('http://', 'ws://');
```
Note that in the current version the `replace('https', 'wss') is useless as it is covered by the `replace('http','ws')`...

Also the `indexOf('robot_windows') seems quite weak, can't we be more specific, e.g., `/robot_windows/`?

Reply:
_Also the indexOf('robot_windows') seems quite weak, can't we be more specific, e.g., /robot_windows/`?_

We can but we need to be sure to increment to also increment the index as we need to keep the first `/`